rita_common/src/middleware.rs

```diff
@@ -1,26 +1,58 @@
 //! This is the Actix-web middleware that attaches the content headers we need for
 //! the client dashboard
 
-use actix_web::http::{header, HttpTryFrom, Method, StatusCode};
-use actix_web::middleware::{Middleware, Response, Started};
-use actix_web::{FromRequest, HttpRequest, HttpResponse, Result};
-use actix_web_httpauth::extractors::basic::{BasicAuth, Config};
-use actix_web_httpauth::extractors::AuthenticationError;
+use actix_web_async::dev::{Service, Transform};
+use actix_web_async::http::header::{
+    Header, ACCESS_CONTROL_ALLOW_HEADERS, ACCESS_CONTROL_ALLOW_ORIGIN,
+};
+use actix_web_async::http::{header, Method, StatusCode};
+use actix_web_httpauth_async::extractors::basic::Config;
+use actix_web_httpauth_async::extractors::AuthenticationError;
+
+use actix_web_async::{dev::ServiceRequest, dev::ServiceResponse, Error};
+use actix_web_httpauth_async::headers::authorization::{Authorization, Basic};
+use futures::future::{ok, LocalBoxFuture, Ready};
+use futures::FutureExt;
 use regex::Regex;
 
-pub struct Headers;
+pub struct HeadersMiddlewareFactory;
+
+impl<S, B> Transform<S, ServiceRequest> for HeadersMiddlewareFactory
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error> + 'static,
+{
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type InitError = ();
+    type Transform = HeadersMiddleware<S>;
+    type Future = Ready<Result<Self::Transform, Self::InitError>>;
 
-impl<S> Middleware<S> for Headers {
-    fn start(&self, _req: &HttpRequest<S>) -> Result<Started> {
-        Ok(Started::Done)
```

Comment:
for this custom headers handler work could you add a comment describing where you got the example?

Reply:
fixed, added a comment to the top of document